frappe/utils/jinja.py

```diff
@@ -65,21 +66,36 @@ def render_template(template, context, is_path=None, safe_render=True):
 	:param safe_render: (optional) prevent server side scripting via jinja templating
 	'''
 
+	if not template:
+		return ""
+
+	import frappe
 	from frappe import _, get_traceback, throw
 	from jinja2 import TemplateError
 
-	if not template:
-		return ""
+	def _render_template():
+		if (is_path or guess_is_path(template)):
+			return get_jenv().get_template(template).render(context)
 
-	if (is_path or guess_is_path(template)):
-		return get_jenv().get_template(template).render(context)
-	else:
 		if safe_render and ".__" in template:
 			throw(_("Illegal template"))
+
 		try:
 			return get_jenv().from_string(template).render(context)
 		except TemplateError:
-			throw(title="Jinja Template Error", msg="<pre>{template}</pre><pre>{tb}</pre>".format(template=template, tb=get_traceback()))
+			throw(
+				title="Jinja Template Error",
+				msg=f"<pre>{template}</pre><pre>{get_traceback()}</pre>"
+			)
+
+	writes_allowed = frappe.flags.read_only
```

Comment:
The meaning of read_only and writes_allowed are completely opposite. Is this intentional?